src/pages/api/stripe/webhook.ts

```diff
@@ -17,8 +33,65 @@ const stripeWebhookHandler = async (
     try {
       event = stripe.webhooks.constructEvent(buf, sig as string, webhookSecret)
 
-      if (event.type === 'checkout.session.completed') {
-        console.log('stripe webhook handler', {event})
+      // Also handle:
+      // - 'customer.subscription.updated'
+      // - 'customer.subscription.deleted'
+      if (event.type === 'customer.subscription.created') {
+        // this types it as a Stripe.Subscription instead of any
+        const stripeSubscription = await stripe.subscriptions.retrieve(
+          event.data.object.id,
+        )
+        const stripeCustomer = await stripe.customers.retrieve(
+          event.data.object.customer,
+        )
+
+        if (!('email' in stripeCustomer)) {
+          // solves a typescript error 🤡
+          throw new Error('No email found for customer')
+        }
+
+        const stripeEmail = stripeCustomer.email
+
+        // we want to get the `co_************` id from CustomerIO which we can
+        // cross-reference with MixPanel to uniquely identify the user.
+        // This correlates to the contact_id of the user in egghead.io
+        let cioCustomer = await cioAxios
+          .get(`/customers`, {
+            params: {email: stripeEmail},
+            headers: {
+              Authorization: `Bearer ${process.env.CUSTOMER_IO_APPLICATION_API_KEY}`,
+            },
+          })
+          .then(({data}: {data: any}) => data.results[0])
+          .catch((error: any) => {
+            console.error(error)
+            return {}
+          })
+
+        // if running sale we would want to check for that but the regular case it's pro or PPP
+        // TODO: need to differentiate between other types of discounts (e.g. sales)
+        let subscriptionType = !stripeSubscription.discount ? 'pro' : 'ppp'
+
+        // stripe puts the plan at the top level of the subscription object
+        // but it isn't on the type so had to do this
+        let subscriptionInterval =
+          stripeSubscription.items.data[0].plan.interval
+
+        mixpanel.track('Subscription Created', {
+          distinct_id: cioCustomer.id,
+          subscriptionType,
+          subscriptionInterval,
+          currentPeriodStart: stripeToMixpanelDataConverter(
+            event.data.object.current_period_start,
+          ),
+          currentPeriodEnd: stripeToMixpanelDataConverter(
+            event.data.object.current_period_end,
+          ),
+        })
```

Comment:
Reviewing the plan, it really seems like we are interested in events and not customer properties for reporting so I moved most of this information here and associated it with the `distinct_id` 

regardless, I preferred longer property names here to avoid ambiguity (subscriptionType, subscriptionInterval, etc)

I didn't see a `Subscription Created` event in the plan doc